taf/tests/test_updater/update_utils.py

```diff
@@ -250,9 +284,54 @@ def _update_expect_error():
     _update_expect_error()
 
     if not expect_partial_update:
-        # the client repositories should not exits
+        # the client repositories should not exist
         for client_repository in client_repos.values():
             if client_repository.path in repositories_which_existed_paths:
                 assert client_repository.path.exists()
             else:
                 assert not client_repository.path.exists()
+
+
+def validate_updated_repositories(
+    library_with_dependencies,
+    origin_dir,
+    client_dir,
+    valid_target_name,
```

Comment:
These could be arrays of repositories that we expect to get updated successfully, and of repositories which should not get updated. We might also want to check the state of referenced auth repos